### PR TITLE
EVG-2501: add repotracker catchups for stepback activation and catchup runs

### PR DESCRIPTION
**Pull request by tychoish**
This is somewhat speculative, and I think we can smoke out the bugs we care about in staging.

Remaining work: 
- I think push events and catchup jobs can interleave awkwardly. 
- degraded mode flags observance 
- run amboy-based catchup for non-legacy runner based checkup. 

This enables: 
- refactoring the catchup function, and running it with slightly different logic so that the normal repotracker operations run via more frequent amboy jobs.

### CODE REVIEW

File: cloud/cloud.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ type CloudManager interface {
 	SpawnHost(h *host.Host) (*host.Host, error)
 
 	// CanSpawn indicates if this provider is capable of creating new instances
-	// with SpawnInstance(). If this provider doesn't support spawning new
+	// with SpawnInstance(). If this provider doesn't support spawning knew
```
Comment (bsamek) [inline cloud/cloud.go]: lol

File: operations/service_runner.go [hunk 2]
```diff
@@ -126,7 +125,13 @@ func startRunnerService() cli.Command {
...
 // Running and executing the offline operations processing.
 
-func startCollectorJobs(ctx context.Context, env evergreen.Environment) {
+func startSystemCronJobs(ctx context.Context, env evergreen.Environment) {
+	// Add jobs to a remote queue for repotracker operations
+	amboy.IntervalQueueOperation(ctx, env.RemoteQueue(), time.Minute, time.Now(), true, units.PopulateActivationJobs(env))
+	amboy.IntervalQueueOperation(ctx, env.RemoteQueue(), 15*time.Minute, time.Now(), true, units.PopulateCatchupJobs())
+	amboy.IntervalQueueOperation(ctx, env.RemoteQueue(), 90*time.Second, time.Now(), true, units.PopulateRepotrackerPollingJobs())
+
+	// add jobs to a local queue every minute.
```
Comment (bsamek) [inline operations/service_runner.go]: What jobs are being added here?

File: repotracker/wrappers.go [hunk 3]
```diff
@@ -47,6 +54,30 @@ func CollectRevisionsForProject(conf *evergreen.Settings, project model.ProjectR
...
+	if !project.Enabled {
+		return errors.Wrap(errProjectDisabled, project.String())
+	}
+
+	tracker, err := getTracker(conf, project)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if err = tracker.activationForProject(project.Identifier); err != nil {
```
Comment (bsamek) [inline repotracker/wrappers.go]: Why is this both in ActivateBuildsForProject and FetchRevisions?

File: operations/service_runner.go [hunk 4]
```diff
@@ -126,7 +125,13 @@ func startRunnerService() cli.Command {
 //
 // Running and executing the offline operations processing.
 
-func startCollectorJobs(ctx context.Context, env evergreen.Environment) {
+func startSystemCronJobs(ctx context.Context, env evergreen.Environment) {
+	// Add jobs to a remote queue for repotracker operations
+	amboy.IntervalQueueOperation(ctx, env.RemoteQueue(), time.Minute, time.Now(), true, units.PopulateActivationJobs(env))
+	amboy.IntervalQueueOperation(ctx, env.RemoteQueue(), 15*time.Minute, time.Now(), true, units.PopulateCatchupJobs())
```
Comment (bsamek) [inline operations/service_runner.go]: How are catchup jobs different from the activation/polling jobs?

File: repotracker/wrappers.go (hunk 3 above)
Comment (tychoish) [reply repotracker/wrappers.go]: This just sees if batchtime has expired and activates any needed tasks in the most recent builds. It's safe to run in both places, and FetchRevisions as such only really runs in the legacy repotracker.

**Pull request merged**